lib/src/main/scala/spinal/lib/hdl/parser/Parser.scala

```diff
@@ -0,0 +1,208 @@
+package fpinscala.parsing
+
+//this code is from https://github.com/fpinscala/fpinscala
```

Comment:
If you're going to copy in other MIT licensed code each file should have an appropriate copyright header.

Reply:
do you mean that i need to copy that `LICENSE` file to each file's head?

Reply:
Yes I think that's sufficient here.  That'll include the license and the author information.

Reply:
and what should i do to `verilog.scala` which is also from fpinscala but i do some change on it